client/src/components/Flashcard/CreateFlashcard.js

```diff
@@ -0,0 +1,177 @@
+import React, { useState } from "react"
+import { Button, Stack, TextField } from "@mui/material"
+
+const CreateFlashcard = ({ deckId }) => {
+  const [values, setValues] = useState({
+    frontImage: "",
+    frontText: "",
+    backImage: "",
+    backText: ""
+  })
+  const [err, setErrs] = useState({
+    frontImageErr: "",
+    frontTextErr: "",
+    backImageErr: "",
+    backTextErr: "",
+    success: false,
+  })
+
+  
+  const { frontImage, frontText, backImage, backText } = values
+  const { frontImageErr, frontTextErr, backImageErr, backTextErr } = err
+
+  const validateFrontImage = () => {
+    const { frontImage } = values;
+      setErrs({...err,
+        frontImageErr:
+          frontImage.length > 3 ? null : 'frontImage must be longer than 3 characters'
+      });
+  }
+
+  const handleFrontImage = event => {
+    setValues({...values, frontImage: event.target.value})
+    validateFrontImage()
+  }
+
+  const validateFrontText = () => {
+    const { frontText } = values;
+      setErrs({...err,
+        frontTextErr:
+          frontText.length > 3 ? null : 'frontText must be longer than 3 characters'
+      });
+  }
+
+  const handleFrontText = event => {
+    setValues({...values, frontText: event.target.value})
+    validateFrontText()
+  }
+
+  const validateBackImage = () => {
+    const { backImage } = values;
+      setErrs({...err,
+        backImageErr:
+        backImage.length > 3 ? null : 'backImage must be longer than 3 characters'
+      });
+  }
+
+  const handleBackImage = event => {
+    setValues({...values, backImage: event.target.value})
+    validateBackImage()
+  }
+
+  const validateBackText = () => {
+    const { backText } = values;
+      setErrs({...err,
+        backTextErr:
+          backText.length > 3 ? null : 'backText must be longer than 3 characters'
+      });
+  }
+
+  const handleBackText = event => {
+    setValues({...values, backText: event.target.value})
+    validateBackText()
+  }
+
+  
+  const handleSubmit = (event) => {
+    event.preventDefault()
+    const card = {
+      frontImage,
+      frontText,
+      backImage,
+      backText
+    }
+    // console.log(card)
+    fetch(`http://localhost:8000/decks/${deckId}/cards`, {
+      method: "POST",
+      headers: {
+          Accept: "application/json",
+          "Content-Type": "application/json",
+          user: "621b050c8e3848f24088b84e"
```

Comment:
Why hard-code this?

Reply:
I have resolved and made it dynamic, kindly re-check the changes. Thank You.